lib/src/source.dart

```diff
@@ -81,7 +75,7 @@ abstract class Source {
   PackageRef parseRef(
     String name,
     Object? description, {
-    required Description containingDescription,
+    required ResolvedDescription containingDescription,
```

Comment:
I kind of wish this migration could have been done in steps.

 1. Change `Description` -> `ResolvedDescription`
 2. Move `hasMultipleVersions`
 3. Adopt usage of `hasMultipleVersions` instead of `is HostedSource`
 4. ..
 5. Support for `tag-pattern`

But I can also see how it would cause a lot of conflicts to do that.

Reply:
Done